File: Data Structures & Algorithms/majority-element/submission-9.java
class Solution {
    public int majorityElement(int[] nums) {
        int numlen = nums.length;
        Map<Integer, Integer> map = new HashMap<>();

        for(int i =0;i<numlen;i++){
            map.put(nums[i], map.getOrDefault(nums[i],0)+1);
        }

        int maxKey = 0;
        int maxValue = 0;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()){
            int k = entry.getKey();
            int v = entry.getValue();
            if(v > maxValue){
                maxValue = v;
                maxKey = k;
            }
        }

        return maxKey;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-9.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        m = 0

        while l <= r:
            m = (l+r) // 2
            if nums[m] == target:
                return m

            if nums[l] <= nums[m]: # 表示左邊是嚴格遞增
                if nums[l] <= target < nums[m]:
                    r = m-1
                else:
                    l = m+1
            else: #右邊嚴格遞增
                if nums[m] < target <= nums[r]:
                    l = m+1
                else:
                    r = m-1

        return -1